### PR TITLE
add unit tests for package cli/command/secret

### PR DESCRIPTION
**Pull request by adshmh**
This PR adds unit tests for all commands under cli/command/secret package (create, list, inspect, and remove). Part of the work on #31217

Signed-off-by: Arash Deshmeh <adeshmeh@ca.ibm.com>

**- What I did**
Added unit tests for all the commands in package cli/command/secret

**- How I did it**
Added unit tests and supporting code (builders + test data files in testdata directory) for all the commands in cli/command/secret package (create, ls, inspect, and remove)

**- How to verify it**
Run the unit tests.
- There are no unit tests for the cli/command/secret package in current version (no test files)

- With this PR:
$ make test-unit
...
ok  	github.com/docker/docker/cli/command/secret	0.069s	coverage: 93.2% of statements
...

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by dnephin — COMMENTED** [review]
Thanks! this looks good, but there are some missing comments causing CI to fail

**Comment by adshmh** [comment]
Thank you for the review. I fixed the comments that were not written correctly and updated the PR.

**Comment by thaJeztah** [comment]
Looks like tests are failing @adshmh 

```
21:11:03 --- FAIL: TestSecretListWithConfigFormat (0.00s)
21:11:03 	assert.go:107: ls_test.go:121: Expected 'foo
21:11:03 		bar label1=label-foo,label2=label-bar
21:11:03 		' got 'foo
21:11:03 		bar label1=label-foo,label2=label-bar
21:11:03 		'%!(EXTRA string=foo 
21:11:03 		bar label2=label-bar,label1=label-foo
21:11:03 		, string=foo 
21:11:03 		bar label2=label-bar,label1=label-foo
21:11:03 		)
21:11:03 Error: error removing secret: foo
21:11:03 Usage:
21:11:03   rm SECRET [SECRET...] [flags]
21:11:03 
21:11:03 Aliases:
21:11:03   rm, remove
21:11:03 
21:11:03 
21:11:03 FAIL
21:11:03 coverage: 93.2% of statements
21:11:03 FAIL	github.com/docker/docker/cli/command/secret	0.020s
```

**Comment by adshmh** [comment]
Thank you for the notification. I think it was due to using multiple labels for test objects (Secret) and the order they appeared in the output (interestingly, all unit-tests passed on my test machine all the time). 
It is fixed now.

**Comment by adshmh** [comment]
Thank you for the notification. I think it was due to using multiple labels for test objects (Secret) and the order they appeared in the output (interestingly, all unit-tests passed on my test machine all the time). 
It is fixed now.

File: cli/command/secret/create_test.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+func TestSecretCreateErrors(t *testing.T) {
+	testCases := []struct {
+		args             []string
+		secretCreateFunc func(swarm.SecretSpec) (types.SecretCreateResponse, error)
+		expectedError    string
+	}{
+		{
+			args:          []string{"too_few"},
+			expectedError: "\"create\" requires exactly 2 argument(s)",
```
Comment (vdemeester) [inline cli/command/secret/create_test.go]: It's a nit but I think we could even just use `"requires exactly 2 argument(s)"` here 👼

File: cli/command/secret/create_test.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		},
+		{args: []string{"too", "many", "arguments"},
+			expectedError: "\"create\" requires exactly 2 argument(s)",
```
Comment (vdemeester) [inline cli/command/secret/create_test.go]: Same here then 😛

File: cli/command/secret/remove_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+func TestSecretRemoveErrors(t *testing.T) {
+	testCases := []struct {
+		args             []string
+		secretRemoveFunc func(string) error
+		expectedError    string
+	}{
+		{
+			args:          []string{},
+			expectedError: "\"rm\" requires at least 1 argument(s).",
```
Comment (vdemeester) [inline cli/command/secret/remove_test.go]: Same here 👼

**Review by vdemeester — APPROVED** [review]
Few comments but LGTM 👼 🐮

File: cli/command/secret/create_test.go (hunk 1 above)
Comment (adshmh) [inline cli/command/secret/create_test.go]: Thank you for the review. The PR is now updated (removed references to the command name from the tests expected errors).

**Review by dnephin — APPROVED** [review]
LGTM

Thanks!

**Pull request merged**